File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c-skills/tsvc_2_s4112__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_s4112_fp64(double *restrict a, const double *restrict b, const int32_t *restrict ip,
                         int64_t LEN_1D, uint8_t *restrict workspace, int64_t workspace_bytes)
{
    (void)workspace;
    (void)workspace_bytes;

    #pragma omp parallel for simd schedule(static)
    for (int64_t i = 0; i < LEN_1D; i++) {
        a[i] = a[i] + b[ip[i]] * 2.0;
    }
}

File: paper_artifacts/experiments/llr9/artifacts/sources/qwen38-fortran/tsvc_2_s4112__submitted.f90
subroutine tsvc_2_s4112_fp64(a, b, ip, len_1d, workspace, workspace_size) bind(C, name="tsvc_2_s4112_fp64")
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: len_1d, workspace_size
  real(c_double), intent(inout) :: a(*)
  real(c_double), intent(in) :: b(*)
  integer(c_int32_t), intent(in) :: ip(*)
  integer(c_int8_t), intent(inout) :: workspace(*)
  integer(c_int64_t) :: i

!$omp parallel do schedule(static)
  do i = 1, len_1d
    a(i) = a(i) + 2.0d0 * b(ip(i))
  end do
end subroutine tsvc_2_s4112_fp64
